### PR TITLE
[ADD][11.0] website_sale_product_reference_displayed: New module

### PR DESCRIPTION
**Pull request by chienandalu**
This module extends the `website_sale` views to display the product's full
display name, with it's product reference included.

cc @Tecnativa TT21256

### CODE REVIEW

File: website_sale_product_reference_displayed/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+This module extends the `website_sale` views to display the product's full
```
Comment (yajo) [inline website_sale_product_reference_displayed/readme/DESCRIPTION.rst]:
```suggestion
This module extends the ``website_sale`` views to display the product's full
```

File: website_sale_product_reference_displayed/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+display name, with it's product reference included.
```
Comment (yajo) [inline website_sale_product_reference_displayed/readme/DESCRIPTION.rst]:
```suggestion
display name, with its product reference included.
```

File: website_sale_product_reference_displayed/views/website_sale_views.xml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+    <template id="products_item" name="Display product reference" inherit_id="website_sale.products_item">
+        <!-- Product title in list -->
+        <xpath expr="//a[@t-field='product.name']" position="attributes">
+            <attribute name="class" add="hidden" separator=" "/>
+        </xpath>
+        <xpath expr="//a[@t-field='product.name']" position="after">
+            <a itemprop="name" t-att-href="keep('/shop/product/%s' % slug(product), page=(pager['page']['num'] if pager['page']['num']&gt;1 else None))" t-att-content="product.display_name" t-field="product.display_name"/>
```
Comment (yajo) [inline website_sale_product_reference_displayed/views/website_sale_views.xml:12]: I'm curious; what's `t-att-content`? :thinking:

**Review by yajo — APPROVED** [review]
Code looks OK.

Views should be de/activable, for multiwebsite support.

File: website_sale_product_reference_displayed/views/website_sale_views.xml (hunk 3 above)
Comment (chienandalu) [reply website_sale_product_reference_displayed/views/website_sale_views.xml:12]: SEO stuff I think

**Comment by chienandalu** [comment]
What do you mean with de/activable?

File: website_sale_product_reference_displayed/views/website_sale_views.xml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        </xpath>
+    </template>
+
+    <template id="recommended_products" name="Display product reference" inherit_id="website_sale.recommended_products">
+        <!-- Product title in recommended products -->
+        <xpath expr="//span[@t-field='alt_product.name']" position="attributes">
+            <attribute name="class" add="hidden" separator=" "/>
+        </xpath>
+        <xpath expr="//span[@t-field='alt_product.name']" position="after">
+            <span t-att-title="alt_product.name" t-field="alt_product.display_name" class="o_text_overflow" style="display: block;" />
```
Comment (Tardo) [inline website_sale_product_reference_displayed/views/website_sale_views.xml]: Here can use:
```suggestion
            <span t-att-title="alt_product.name" t-field="alt_product.display_name" class="o_text_overflow show" />
```

File: website_sale_product_reference_displayed/views/website_sale_views.xml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        </xpath>
+        <xpath expr="//span[@t-field='product.name']" position="after">
+            <span t-field="product.display_name"/>
+        </xpath>
+        <!-- Product Title -->
+        <xpath expr="//h1[@t-field='product.name']" position="attributes">
+            <attribute name="class" add="hidden" separator=" "/>
+        </xpath>
+        <xpath expr="//h1[@t-field='product.name']" position="after">
+            <h1 itemprop="name" t-field="product.display_name">Product Name</h1>
```
Comment (Tardo) [inline website_sale_product_reference_displayed/views/website_sale_views.xml]: This print "Product Name" if 'display_name' is empty?

**Review by Tardo — APPROVED** [review]
LGTM :+1: Only a non-blocking suggestion

**Comment by yajo** [comment]
> What do you mean with de/activable?

**Comment by pedrobaeza** [comment]
If I don't remember bad, this is the option `customize_show=True` in the template

File: website_sale_product_reference_displayed/views/website_sale_views.xml (hunk 5 above)
Comment (chienandalu) [reply website_sale_product_reference_displayed/views/website_sale_views.xml]: Yeah, I just copied the original from Odoo, but it's useless indeed

File: website_sale_product_reference_displayed/views/website_sale_views.xml (hunk 4 above)
Comment (chienandalu) [reply website_sale_product_reference_displayed/views/website_sale_views.xml]: Ok, same as below. Fixed. Thanks for the suggestion :)

**Comment by chienandalu** [comment]
> customize_show=True

Ah, yeah. Cool. I didn't remember that one :)

**Review by yajo — APPROVED** [review]

**Pull request merged**